core/conn/jdbc_type2/native/CSrvrStmt.cpp

```diff
@@ -180,12 +241,88 @@ SQLRETURN SRVR_STMT_HDL::Prepare(const SQLValue_def *inSqlString, short inStmtTy
 
     strncpy((char *)sqlString.dataValue._buffer, (const char *)inSqlString->dataValue._buffer, inSqlString->dataValue._length);
     sqlString.dataValue._buffer[inSqlString->dataValue._length] = '\0';
+
+    sqlStringLen = inSqlString->dataValue._length;
+    MEMORY_ALLOC_ARRAY(sqlStringText, char, sqlStringLen+1);
+    strncpy((char *)sqlStringText, (const char *)inSqlString->dataValue._buffer, sqlStringLen);
+    sqlStringText[inSqlString->dataValue._length] = '\0';
+
     stmtType = inStmtType;
     holdability = inHoldability;
 
     CLI_DEBUG_RETURN_SQL(PREPARE(this));
 }
 
+SQLRETURN SRVR_STMT_HDL::Execute(const char *inCursorName, IDL_long inInputRowCnt, IDL_short inSqlStmtType, 
+        const SQLValueList_def *inValueList, 
+        short inSqlAsyncEnable, Int32 inQueryTimeout)
+{
```

Comment:
The method signature of this method and the one below looks same. If so, it is not clear how the compiler decides to resolve this method.